src/Renci.SshNet/SftpClient.cs

```diff
@@ -2106,7 +2116,7 @@ private void InternalUploadFile(Stream input, string path, Flags flags, SftpUplo
 
             var handle = _sftpSession.RequestOpen(fullPath, flags);
 
-            ulong offset = 0;
+            ulong offset = (ulong)input.Position;
```

Comment:
File upload: Set the initial offset of the remote file to the same as the local file